sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c

```diff
@@ -136,6 +136,50 @@ ng_ipv6_nc_t *ng_ipv6_nc_get(kernel_pid_t iface, const ng_ipv6_addr_t *ipv6_addr
     return NULL;
 }
 
+ng_ipv6_nc_t *ng_ipv6_nc_get_next(ng_ipv6_nc_t *prev)
+{
+    if (prev == NULL) {
+        prev = ncache;
+    }
+    else {
+        prev++;     /* get next entry */
+    }
+
+    while (prev - ncache < NG_IPV6_NC_SIZE) {   /* while not reached end */
```

Comment:
changing the inequality to `prev < NG_IPV6_NC_SIZE + ncache` might be preferable. In this case the cpu might cache the right hand in one of its registers, hence computing it only once.